### ISSUE TITLE
Misleading message in case remote machine is out of disk space

### ISSUE THREAD

**Issue by avaika** [ansible/ansible#25401]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
Core component.

##### ANSIBLE VERSION
```
$ ansible --version
ansible 2.2.1.0
  config file = /home/user/hosting/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
```
$ cat ansible.cfg
[defaults]
#host_key_checking = False
#log_path = /var/log/ansible.log
executable = /bin/bash
#vault_password_file = /path/to/vault_password_file
ansible_managed = Ansible managed: {file} modified on %Y-%m-%d %H:%M:%S by {uid} on {host}
callback_whitelist = profile_tasks

[ssh_connection]
pipelining = True
ssh_args = -C -o ControlMaster=auto -o ControlPersist=120s
```

##### OS / ENVIRONMENT
N/A

##### SUMMARY
The error message in case there is no space for temporary ansible files on remote machine is misleading. I had to spent quite a while to figure out a real issue. 
It would be convenient to change it somehow, so people won't waste time to check connectivity issues, google on this topic and so on. 

PS. While it might be a solution to connect as root on remote machine (there are a lot of advices to do so on forums), it won't solve the real issue and sometimes won't work (e.g. if root 5% quota (if configured) is also eaten).

##### STEPS TO REPRODUCE
1. Find the server with no space left (e.g. create a huge file with fallocate -l 1000G test.img
 or put any value instead of 1000G)
2. Try to run any playbook / module (even a simple ping) against this machine
...
3. Here you are. The error message which forces people to check connectivity, local issues, google and have some fun :)

##### EXPECTED RESULTS
Error which will at least give some kind of hint, that there is an issue with the space on remote machine.

##### ACTUAL RESULTS
```
$ ansible -i hosts server -m ping -vvv
Using /home/user/hosting/ansible.cfg as config file
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/core/system/ping.py
<server> ESTABLISH SSH CONNECTION FOR USER: None
<server> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=120s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/user/.ansible/cp/ansible-ssh-%h-%p-%r server '/bin/bash -c '"'"'/usr/bin/python && sleep 0'"'"''
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "<stdin>", line 136, in <module>
NameError: name 'temp_path' is not defined

server | FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_name": "ping"
    }, 
    "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 136, in <module>\nNameError: name 'temp_path' is not defined\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE"
}
```

**Comment by gundalow** [comment]
@abadger Did you recently do some work on Ansiballz and low diskspace and/or `temp_path`?

**Comment by abadger** [comment]
Yes.  @avaika are you able to test this PR: https://github.com/ansible/ansible/pull/25394 ?

**Comment by abadger** [comment]
PR has been merged to devel and stable-2.3 branches so  it will be in the next release of 2.3.x and in 2.4.0.

needs_info

**Comment by avaika** [comment]
@abadger . I tested. Looks perfect. Here is the output:
```
server | UNREACHABLE! => {
    "changed": false, 
    "msg": "mkdir: cannot create directory '/home/user/.ansible/tmp/ansible-tmp-1496866547.5-280913079334864': No space left on device\n", 
    "unreachable": true
}
```
So I'll close the ticket.
Thank you for the change.
